libraries/ESP8266WebServer/src/detail/RequestHandler.h

```diff
@@ -14,6 +16,16 @@ class RequestHandler {
 
 private:
     RequestHandler* _next = nullptr;
+
+protected:
+    std::vector<String> pathArgs;
+
+public:
+    String pathArg(unsigned int i) { 
+        if (i < pathArgs.size())
```

Comment:
If the index is out of bounds, do we really want to silently fail? Or is it a programming error? If the latter, there should be an assert checking the index. Then, due to the assurance, the return type can be a const reference instead of by value.